src/components/Button.jsx

```diff
@@ -0,0 +1,10 @@
+// @flow
+import React from 'react';
+
+import type { Element } from 'react';
+
+const Button = ({
```

Comment:
Great use of generic value, but this might be too generic, ha ha.  Something more like CounterButton might be better. When we get into styled components you'll see why.